x-pack/packages/security-solution/ecs_data_quality_dashboard/impl/data_quality_panel/data_quality_details/indices_details/pattern/index_check_flyout/check_fields_tabs/index.tsx

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useMemo, useState } from 'react';
+import {
+  EuiBadge,
+  EuiButtonGroup,
+  EuiButtonGroupProps,
+  EuiFlexGroup,
+  EuiSpacer,
+  EuiToolTip,
+} from '@elastic/eui';
+import styled, { StyledComponent } from 'styled-components';
+
+import { CheckFieldsTab, CheckFieldsTabId } from './types';
+
+const StyledTabFlexGroup = styled(EuiFlexGroup)`
+  width: 100%;
+`;
+
+const StyledTabFlexItem = styled.div`
+  text-overflow: ellipsis;
+  white-space: nowrap;
+  overflow: hidden;
+`;
+
+const StyledBadge = styled(EuiBadge)`
+  text-align: right;
+  cursor: pointer;
+`;
+
+interface CheckFieldsSingleButtonGroupProps {
+  onChange: (id: string) => void;
+  idSelected: CheckFieldsTabId;
+  options: EuiButtonGroupProps['options'];
+  legend: 'Check fields tab group';
+  buttonSize: 'compressed';
+  color: 'primary';
+}
+
+export interface Props {
+  tabs: CheckFieldsTab[];
+  renderButtonGroup: (
+    props: CheckFieldsSingleButtonGroupProps
+  ) => React.ReactElement<
+    EuiButtonGroupProps,
+    StyledComponent<typeof EuiButtonGroup, {}, {}, never> | typeof EuiButtonGroup
+  >;
+}
+
+const CheckFieldsTabsComponent: React.FC<Props> = ({ tabs, renderButtonGroup }) => {
+  const checkFieldsTabs = useMemo(
+    () =>
+      tabs.map((tab) => ({
+        id: tab.id,
+        name: tab.name,
+        append: <StyledBadge color={tab.badgeColor ?? 'hollow'}>{tab.badgeCount ?? 0}</StyledBadge>,
+        content: tab.content ?? null,
+        disabled: Boolean(tab.disabled),
+        ...(tab.disabled && { disabledReason: tab.disabledReason }),
+      })),
+    [tabs]
+  );
+
+  const [selectedTabId, setSelectedTabId] = useState<CheckFieldsTabId>(() => checkFieldsTabs[0].id);
```

Comment:
NIT:
```suggestion
  const [selectedTabId, setSelectedTabId] = useState<CheckFieldsTabId>(tabs[0].id);
```

Reply:
I was initially wondering what was the need for the function. But the default tab is always the same, we could even do `useState<CheckFieldsTabId>(INCOMPATIBLE_TAB_ID);` for clarity. But as you prefer, it's not an important thing.

Reply:
yes since there is no clear requirement whether it should be incompatible tab or first tab from the left. Not that we discussed. I just went with the latter because it detaches from specific id and allows flexibility for if you decide to reshuffle later or if you want to test in your tests without incompatible tab for some reason idk.